### PR TITLE
Improved sorting UX for topic pages

### PR DESCRIPTION
**Pull request by Will-Howard**
See designs here: https://www.figma.com/file/OD4vJlLuWriHqG1WJXY4LV/Q1.1-2023?node-id=1006-5968&t=D7xPICXN7FBuLsrk-0

Screenshot 2023-03-29 at 15 28 22
Screenshot 2023-03-29 at 15 28 30
Screenshot 2023-03-29 at 15 28 40
Screenshot 2023-03-29 at 15 28 46
Screenshot 2023-03-29 at 15 28 52
Screenshot 2023-03-29 at 15 29 00
Screenshot 2023-03-29 at 15 29 49
Screenshot 2023-03-29 at 15 29 58
Screenshot 2023-03-29 at 15 30 48
Screenshot 2023-03-29 at 15 31 39

cc @agnesstenlund 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204270942090967) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/tagging/SubscribeButton.tsx [hunk 1]
```diff
@@ -82,22 +223,56 @@ const SubscribeButton = ({
...
       </Button>
-    </LWTooltip>
-    {showNotificationBell && currentUser && <TagNotificationSettings
-      tag={tag}
-      userTagRel={userTagRel}
-      currentUser={currentUser}
-      isFrontpageSubscribed={!!isSubscribed}
-    />}
-  </div>
+      {/* TODO add analytics back in */}
```
Comment (Will-Howard) [inline packages/lesswrong/components/tagging/SubscribeButton.tsx:258]: I am planning to do this before merging

File: packages/lesswrong/components/tagging/subforums/SubforumListSettings.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/tagging/subforums/SubforumListSettings.tsx:1]: I deleted a few unused files while I was here

**Comment by Will-Howard** [comment]
Reassigning to @s-cheng because JP just reviewed another thousand line PR of mine (sorry Sarah)

**Comment by agnesstenlund** [comment]
Really really nice!! 👏 excited about this. I only have small visual things (fixed a few ones already, see commit)

Some things I didn't fix:

- [ ] Noticed the card/list icon are a bit different in size and think this is my bad–tweaked the icons and exported them again, would be great if you could help me swap them:

card
list

- [ ] The left/right padding on this button is a bit much when hovered
Screenshot 2023-03-30 at 12 08 41

- [ ] The subscribe text should still be white in dark mode:

Screenshot 2023-03-30 at 12 34 15

- [ ] Dropdown shadow doesn't seem adjusted to the dark mode
Screenshot 2023-03-30 at 12 34 54


- [ ] Smallest nit-pick ever, but I like it better when the top-margin in the dropdown is the same as the left/right, like on the "Write new" dropdown:

Screenshot 2023-03-30 at 12 32 39

On the other dropdowns there is slightly more space on the top:
Screenshot 2023-03-30 at 12 32 43

**Comment by jpaddison3** [comment]
> Reassigning to \@s-cheng because JP just reviewed another thousand line PR of mine (sorry Sarah)

🥺 🙏

**Comment by Will-Howard** [comment]
@agnesstenlund thanks for the review!

> * [ ]  The subscribe text should still be white in dark mode:
> * [ ]  Dropdown shadow doesn't seem adjusted to the dark mode

For these two things it would be nice to change for the whole site rather than just this specific case so I'll make tickets to come back to them (maybe in this PR, but probably in a different one). For the subscribe text one do you want this to be the case for all "primary" styled buttons in dark mode?

**Comment by agnesstenlund** [comment]
> For these two things it would be nice to change for the whole site rather than just this specific case so I'll make tickets to come back to them (maybe in this PR, but probably in a different one). For the subscribe text one do you want this to be the case for all "primary" styled buttons in dark mode?

Great shout! Agree.  And yep, I think the text should be white for all primary buttons in dark mode (this is the most common, see github for example 👇)
Screenshot 2023-03-31 at 14 11 02

File: packages/lesswrong/components/tagging/WriteNewButton.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      outline: "none",
+    },
+    '&:hover': {
+      backgroundColor: theme.palette.grey[250],
+      color: theme.palette.grey[1000],
+    }
+  },
+})
+
+export const taggedPostWording = taggingNameIsSet.get() ? `posts with this ${taggingNameSetting.get()}` : "posts with this tag"
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/WriteNewButton.tsx:74]: unused?

File: packages/lesswrong/server/migrations/20230314T152203.add_backfilled.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ export const up = async ({db}: MigrationContext) => {
   }
   
   await addField(db, TagRels, "backfilled")
+  await db.any(`UPDATE "Users" SET "subforumPreferredLayout" = 'card' WHERE "subforumPreferredLayout" = 'feed';`)
```
Comment (s-cheng) [inline packages/lesswrong/server/migrations/20230314T152203.add_backfilled.ts:12]: Leftover code?

File: packages/lesswrong/components/tagging/subforums/TagSubforumPage2.tsx [hunk 4]
```diff
@@ -369,19 +341,14 @@ const TagSubforumPage2 = ({classes}: {
...
-        </RightSidebarColumn>
-      </div>
+      <RightSidebarColumn
+        titleComponent={titleComponent}
+        bannerImageId={tag.bannerImageId}
+        headerComponent={headerComponent}
+        sidebarComponents={rightSidebarComponents[tab]}
+      >
+        {tabComponents[tab]}
+      </RightSidebarColumn>
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/subforums/TagSubforumPage2.tsx:351]: Did this component used to be just the right sidebar, and now contains basically all the main page contents? If so, would be nice to rename it

File: packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx [hunk 5]
```diff
@@ -1,92 +1,111 @@
...
+    [theme.breakpoints.down('sm')]: {
+      height: TITLE_HEIGHT_MOBILE,
+    }
+  },
+  translucentOverlay: {
+    position: 'absolute',
+    zIndex: Z_OVERLAY,
+    width: '100%',
+    height: TITLE_HEIGHT_DESKTOP,
+    backgroundColor: 'rgba(0,0,0,0.4)', // hardcode overlay to keep it the same in light and dark mode
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx:60]: I think the right thing to do is add this new color to the palette and set it to be the same in light and dark mode

File: packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx [hunk 6]
```diff
@@ -98,30 +117,89 @@ export const styles = (theme: ThemeType): JssStyles => ({
...
+   *    as `position: absolute` to make them display on top of each other
+   * 2. The downside of doing the `position: relative`/`position: absolute` trick is that there is
+   *    no way for the two divs to know each others height from pure CSS. Additionally, the cropping
+   *    in CloudinaryImage2 relies on knowing the height in code, so it's ~necessary to set the height
+   *    in code.
+   *    This can't be done until after the page has rendered once, which would cause a layout
+   *    shift. To fix this I have made it so the banner image starts off slightly oversized but is hidden
+   *    behind the next element by using z-indices. Once the page has rendered once, the height gets
+   *    updated to exactly the right value which fixes the cropping (this is fairly discreet)
+   */
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx:152]: I found this kind of hard to follow, but it's also the end of the day on Friday so I'm not the most focused. So you're saying you wanted to use `background-image` but are not currently using that?

All the hard-coding in this file isn't my favorite, but it does seem to work well so I'm happy to keep it for now. It doesn't feel necessary for me to totally dive into this. If you'd like me to do so in the future, I can make some time for that. :) For now I'll just say, another way to make the text stick to the bottom of the image is to put the text in a flex container in the column direction (I didn't read closely enough to know if this would be helpful for you, but thought I'd mention it just in case).

File: packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx [hunk 7]
```diff
@@ -98,30 +117,89 @@ export const styles = (theme: ThemeType): JssStyles => ({
...
+  };
+
+  useEffect(() => {
+    updateElementsHeight();
+    window.addEventListener('resize', updateElementsHeight);
+  
+    return () => {
+      window.removeEventListener('resize', updateElementsHeight);
+    };
+  }, []);
```
Comment (s-cheng) [inline packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx:175]: Perhaps worth using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) instead?

**Review by s-cheng — APPROVED** [review]
This looks soo nice, super excited to have it on the site! :D

I don't think there's anything blocking, but please read my comments.

**Comment by s-cheng** [comment]
@Will-Howard @agnesstenlund Oh also, I think we wanted to use the dropdown on the home page for the topics tabs too - I think that makes sense to do in a separate PR, but just want to make sure we are tracking that somewhere?

File: packages/lesswrong/lib/filterSettings.ts [hunk 8]
```diff
@@ -172,11 +172,11 @@ export const filterModeIsSubscribed = (filterMode: FilterMode) => filterMode ===
...
  * subscription
  */
-export const useSubscribeUserToTag = (tag: TagBasicInfo) => {
+export const useSubscribeUserToTag = (tag?: TagBasicInfo) => {
   const { filterSettings, setTagFilter } = useFilterSettings()
   
-  const filterSetting = filterSettings.tags.find(ft => ft.tagId === tag._id)
-  const isSubscribed = filterSetting && (filterModeIsSubscribed(filterSetting.filterMode))
+  const filterSetting = filterSettings.tags.find(ft => tag && ft.tagId === tag._id)
+  const isSubscribed = !!(filterSetting && (filterModeIsSubscribed(filterSetting.filterMode)))
```
[the remark this answers is not in the document]
Comment (Will-Howard) [reply packages/lesswrong/lib/filterSettings.ts:179]: This is the thing returned from the previous line, not `filterSettings` (1 char difference). I've renamed it to `tagFilterSetting` to make it a bit clearer

File: packages/lesswrong/components/tagging/subforums/TagSubforumPage2.tsx (hunk 4 above)
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/subforums/TagSubforumPage2.tsx:351]: I've renamed it, perhaps controversially, to `SubforumLayout`. This is just for consistency with the rest of the code where `isSubforum` is the flag that decides whether this page is shown

File: packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx (hunk 6 above)
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx]: Thanks, I'll remember the flex trick for next time. I did try a lot of different approached here and ended up with this fairly complex one, which I'm not that happy with. But it does do the job at least

File: packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx (hunk 7 above)
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/subforums/RightSidebarColumn.tsx]: Not going to change but I'll try and remember for next time
